src/server/spectrum_files.R

```diff
@@ -11,26 +12,40 @@ spectrumFiles <- function(input, output, state) {
     })
 
     shiny::observeEvent(input$spectrumFile, {
+        state$newCountry <- FALSE
         inFile <- input$spectrumFile
+        state$multipleCountryError <- FALSE
+        state$newCountry <- FALSE
         if (!is.null(inFile)) {
-            dataSet = list(
-                name = inFile$name,
-                data = first90::prepare_inputs(inFile$datapath)
-            )
-            state$dataSets <- c(state$dataSets, list(dataSet))
-            # TODO: Throw error if data sets after the first do not match the country of the first data set
-            state$country <- read_country(inFile$datapath)
+
+            newCountry <- read_country(inFile$datapath)
+
+            if (is.null(state$country) || !state$anyDataSets()){
+                state$newCountry <- TRUE
+                state$country = newCountry
+
+                dataSet = list(name = inFile$name,
+                                data = first90::prepare_inputs(inFile$datapath))
+
+                state$dataSets <- c(state$dataSets, list(dataSet))
+            }
+            else if (state$country != newCountry){
+                state$multipleCountryError <- TRUE
+                shinyjs::reset("spectrumFile")
+            }
```

Comment:
I don't think this looks right. In the case where you upload a second file of the same country, we're now no longer doing anything with it, right? Whereas we should be adding it to `state$dataSets`.

Reply:
Yes you're right